### ISSUE TITLE
Days Gone 2-3 FPS

### ISSUE THREAD

**Issue by dreamhunt** [doitsujin/dxvk#2065]
Hello,
I get just 2-3 FPS with this game. It starts normally but as soon as it reaches the main menu and it slows down to 2-3 FPS. I tried to start a new game but I experience the same and it's not just in the main menu. I tried different Wine/Proton/ProtonGE versions and video drivers to no avail. I tried to get an apitrace but the game freezes at start. I installed Windows but it seems I can't get an apitrace even on this sh*t. 
I think it's a DXVK version for I tied the game with DXVK on Windows and I experience exactly the same 2-3 FPS problem. Without DXVK it's running just fine.

### Software information
Name of the game, settings used etc.

### System information
- GPU: GTX 1080Ti
- Driver: 455.50.19
- Wine version: 6.8
- DXVK version: 1.8.1

### Log files
- d3d11.log:
[DaysGone_d3d11.log](https://github.com/doitsujin/dxvk/files/6503758/DaysGone_d3d11.log)

- dxgi.log:
[DaysGone_dxgi.log](https://github.com/doitsujin/dxvk/files/6503760/DaysGone_dxgi.log)

**Comment by danielhurt** [comment]
Sorry I am not able to solve your issue. Do you mind telling me where you found the dll files for Days Gone? They're not in the steam folder.

**Comment by dreamhunt** [comment]
What dll files?

**Comment by danielhurt** [comment]
The d3d11 & dxgi files in the Days Gone main folder I'm supposed to replace with the dxvk ones

**Comment by dreamhunt** [comment]
You don't have to replace anything for they are not there. Just put them in the game's exe folder. Look at the DXVK wiki and don't ruin my thread please.

**Comment by dreamhunt** [comment]
Yes, I know apitrace can make things very slow. I provided a lot apitraces but this indeed freezes the game.

**Comment by doitsujin** [comment]
can you try `perf top` or something to see where it's spending all that time? 2-3 FPS can't just be some random dxvk slowness, the game has to do something really really really dumb for that to happen.

**Comment by dreamhunt** [comment]
Yep, the GPU is extremely low utilized:
https://i.imgur.com/eaSWhbE.png
Here you are a perf top log:
[perf.log](https://github.com/doitsujin/dxvk/files/6504808/perf.log)

**Comment by Oschowa** [comment]
The perf top output might be more useful if you limit it to the game process with i.e. `perf top -s comm,dso,symbol -p $(pgrep gamename.exe)` and use a dxvk build that is not stripped. You can use this build of dxvk master if you don't want to build one yourself: https://drive.google.com/file/d/1LqPoQO5Omz_4z92s8Hmmof5ZZ_y_ShkL/view?usp=sharing

**Comment by dreamhunt** [comment]
Thank you, here is the new log :)
[perf.log](https://github.com/doitsujin/dxvk/files/6506866/perf.log)

**Comment by doitsujin** [comment]
looking at all the memset/memcpy spam there, does `d3d11.apitraceMode = True` help?

**Comment by dreamhunt** [comment]
That's great, thanks.
I can confirm `d3d11.apitraceMode = True` doesn't help on my end either.

**Comment by doitsujin** [comment]
Looking at the trace, it doesn't really do anything special, and in fact it replays at ~27 FPS here once it reaches the menu, indicating that the problem is somewhere in the game code rather than DXVK.

Could try vendor ID overrides or something, but not really sure what to do here since we don't seem to be doing anything wrong.

**Comment by dreamhunt** [comment]
But it's strange the problem, also exists on Windows with DXVK.

**Comment by dreamhunt** [comment]
I tried:
`dxgi.customDeviceId = 10de
dxgi.customVendorId = 1c06`
`dxgi.customDeviceId = e366
dxgi.customVendorId = 1002`
and:
`allowMapFlagNoWait`
But nothing helps.

**Comment by Ryusennin** [comment]
Out of curiosity, I tried on Win10 and Win7 (because why not) and it does the exact same thing on both systems. The title screen with the motorbike is crawling at 2 fps, although the CPU and GPU barely reach 2%.

**Comment by Joshua-Ashton** [comment]
Would you be able to see if https://github.com/doitsujin/dxvk/commit/eac94823c82b77530ea2991c83f05dc81711c3e5 has any impact on the game's perf? Artifacts should be available here. https://github.com/doitsujin/dxvk/runs/2630514955

@ZereoX @dreamhunt

**Comment by gofman** [comment]
I justed debugged the game and yes, it needs GetFrameStatistics and GetLastPresentCount to return some about correct values (I didn't try with the referenced commit though yet, used my own WIP debug changes to check).

**Comment by dreamhunt** [comment]
@Joshua-Ashton, thanks but I really don't know how to use this. I can compile DXVK, I can apply patches but I don't know how to use this commit to apply it to the source in order to compile it :(

**Comment by dreamhunt** [comment]
Can you give me the dlls with this commit to try it or explain to me how to get the changes in ordr to compile them?

**Comment by Joshua-Ashton** [comment]
I linked a build in my message.

**Comment by dreamhunt** [comment]
Unfortunately it doesn't help:
![Screenshot_20210520_173824](https://user-images.githubusercontent.com/9913684/118998600-3fb32180-b992-11eb-93b8-92d26cf2f9f4.png)

**Comment by ZereoX** [comment]
@Joshua-Ashton Same results using the build artifacts linked above.

**Comment by Joshua-Ashton** [comment]
You'll need to set `WINEDLLOVERRIDES=dxgi=n`, sorry I did not mention that.

I will also get the game and test it myself.

**Comment by gofman** [comment]
@Joshua-Ashton For reference, here is the patch on top of Proton's dxgi which help here (100+ fps on menu).
[0001-dxgi-Return-some-sensible-values-from-proxy_swapchai.txt](https://github.com/doitsujin/dxvk/files/6516804/0001-dxgi-Return-some-sensible-values-from-proxy_swapchai.txt)

**Comment by Joshua-Ashton** [comment]
Will look at this again later today once I finish rehearsals for a show I am in :)

**Comment by Kron4ek** [comment]
@gofman Thank you, this patch indeed solves the performance problem in this game.

**Comment by AntonioR-dev** [comment]
Could someone link patched files? I'm here because of the same issue but i don't understand how to patch

**Comment by Kron4ek** [comment]
Patched dxgi for Proton 6.3-4. Replace dxgi.dll.so (but first backup it) in steamapps/common/Proton 6.3/wine/lib64/wine.

[dxgi.dll.so.zip](https://github.com/doitsujin/dxvk/files/6517898/dxgi.dll.so.zip)

**Comment by dreamhunt** [comment]
Then use Linux ;)

**Comment by danielhurt** [comment]
> Then use Linux ;)
> Thank you, Kron4ek!

Thanks, worked for me.

**Comment by PlayX83** [comment]
it seems there is a little bug with the patch. if I set vsync to 60 the cap is 30, if I set it to 30 the cap is 15

**Comment by doitsujin** [comment]
Can you test https://github.com/doitsujin/dxvk/tree/dxgi-memes ?

**Comment by dreamhunt** [comment]
Yes! And it's working much faster than dxgi.dll.so. With your build the game is running over 240 FPS and with the patched dxgi.dll.so ~75-80!

**Comment by dreamhunt** [comment]
Opps, the performance difference is due Wine versions. With Proton and newest Wine versions it's slow but with Wine-Staging-Tkg 5.15 it flies! Anyway, your build is working great. Thank you!

**Comment by danielhurt** [comment]
Any fixes for Windows?

**Comment by DadSchoorse** [comment]
the branch should work on windows

**Comment by AntonioR-dev** [comment]
> the branch should work on windows

Could you please add some guide to compile dlls on windows or maybe add the compiled dlls? I’mnot that practical with this stuff, i just wanted to play days gone with amd gpu :)

**Comment by Joshua-Ashton** [comment]
There are builds for every commit if you click the checkmark for it.

**Comment by DadSchoorse** [comment]
or if you want the direct link: https://github.com/doitsujin/dxvk/suites/2793793675/artifacts/62048006

**Comment by danielhurt** [comment]
What I was looking for, thanks.

**Comment by Doyyyy** [comment]
> or if you want the direct link: https://github.com/doitsujin/dxvk/suites/2793793675/artifacts/62048006

Did this and allthough my fps are way better now in all areas, they immediately drop down to 10-20 every few seconds as soon as I start moving. Makes the game unplayable for me

**Comment by dreamhunt** [comment]
I tried Proton and Wine and I don't have such problem. The game is running very well.

**Comment by K0bin** [comment]
@Doyyyy Are you sure what you're seeing isn't just shader compilation stutter? In that case it should get better as you keep playing.

**Comment by Doyyyy** [comment]
Oh yeah that could be it. I'll test that.

**Comment by AntonioR-dev** [comment]
I am testing this on windows and so far I can tell that everything is fine except for the fact that the game now does not recognize HDR monitor. I don't know if this is an issue or it is normal behaviour

**Comment by Doyyyy** [comment]
@K0bin I think that was really the problem. Left the game open for a bit and it's way better now. Still a few stutters so I'm gonna keep it open a bit longer, but already way better. Thanks for the help!

**Comment by K0bin** [comment]
@AntonioR-dev DXVK does not support HDR.

**Comment by AntonioR-dev** [comment]
@K0bin strange, because I played AC Origins with dxvk and hdr. Well, no problem, thank you all!

**Comment by K0bin** [comment]
Probably placebo.

**Comment by cinatic** [comment]
i can't get it to work wether with tkg nor GE proton tried several version from 5 to latest, replacing only the .so file and also using the whole latest dxvk CI build. With the latest dxvk i get decent fps in the menu and cutscenes but as soon as gameplay starts it drops to 0-1 fps, changing the graphic settings has no influence

**Comment by axet** [comment]
I'm having same issues here.

I've tried to run different versions of the game: steam and gog - boot, same issue.

When game starts I got 60FPS stable. Intro showing 60FPS. Menu showing 60FPS. As soon game starts I get 1FPS. When I press escape (with game screen in background) it shows 60FPS. If I get back to the game - 1FPS again!

Makes no sense. Like game engine doing something really wrong.

I tried to run game without DXVK but WINED3D showing 4FPS in menu, and broken colors. It should be reopened.

**Comment by axet** [comment]
I compiled and install ntsync kernel module and wine-tkg-ntsync. Games runs at full speed gives me 70FPS in the game with maxiumum graphical settings on my hardware (12100F, RX6600).
